compiler/rustc_passes/src/check_attr.rs

```diff
@@ -717,6 +717,39 @@ impl CheckAttrVisitor<'tcx> {
         true
     }
 
+    /// Checks that `doc(test(...)` attribute contains only valid attributes. Returns `true` if
+    /// valid.
+    fn check_test_attr(&self, meta: &NestedMetaItem) -> bool {
+        let mut is_valid = true;
+        if let Some(metas) = meta.meta_item_list() {
+            for i_meta in metas {
+                match i_meta.name_or_empty() {
+                    sym::attr | sym::no_crate_inject => {}
+                    name => {
+                        self.tcx
+                            .sess
+                            .struct_span_err(
+                                i_meta.span(),
+                                &format!("unknown `doc(test)` attribute `{}`", name,),
+                            )
+                            .emit();
+                        is_valid = false;
+                    }
+                }
+            }
+        } else {
+            self.tcx
+                .sess
+                .struct_span_err(
+                    meta.span(),
+                    "`#[doc(test(...)]` is supposed to be a list attribute",
```

Comment:
Might be clearer to say something like this:

```suggestion
                    "`#[doc(test(...)]` takes a list of attributes",
```

since people may not know what a list attribute is.

Reply:
:+1: